src/web/Settings.re

```diff
@@ -239,6 +265,71 @@ module Update = {
           ...settings,
           explainThis,
         };
+      | Assistant(u) =>
+        switch (u) {
+        | UpdateChatStatus => {
+            ...settings,
+            assistant: {
+              ...settings.assistant,
+              ongoing_chat: !settings.assistant.ongoing_chat,
+            },
+          }
+        | SwitchMode(mode) => {
+            ...settings,
+            assistant: {
+              ...settings.assistant,
+              mode,
+            },
+          }
+        | ToggleHistory => {
+            ...settings,
+            assistant: {
+              ...settings.assistant,
+              show_history: !settings.assistant.show_history,
+            },
+          }
+        | SetLLM(llm_id) =>
+          {
+            Store.Generic.save("MODEL", llm_id);
+          };
+          settings;
+        | ToggleAPIKeyVisibility => {
+            ...settings,
+            assistant: {
+              ...settings.assistant,
+              show_api_key: !settings.assistant.show_api_key,
+            },
+          }
+        | SetAPIKey(api_key) =>
```

Comment:
same as above; this isn't just a settings toggle; we shouldn't be doing schedule_action here